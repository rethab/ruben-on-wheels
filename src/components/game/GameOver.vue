<template>
  <v-card>
    <v-card-title>Game Over</v-card-title>
    <v-card-text
      >Unfortunately, nobody is going to visit Ruben this time ðŸ¥²</v-card-text
    >
    <v-card-actions>
      <ActionButton text="Play Again" link="/init" />
    </v-card-actions>
  </v-card>
</template>

<script setup lang="ts">
import ActionButton from "@/components/misc/ActionButton.vue";
</script>
