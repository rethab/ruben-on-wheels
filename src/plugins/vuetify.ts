import "vuetify/styles";
import { createVuetify } from "vuetify";
import { aliases, mdi } from "vuetify/iconsets/mdi";
import * as components from "vuetify/components";
import * as directives from "vuetify/directives";
import "@mdi/font/css/materialdesignicons.css";

export default createVuetify({
  icons: { defaultSet: "mdi", aliases, sets: { mdi } },
  components,
  directives,
});

// Export for test.
export { components, directives };
