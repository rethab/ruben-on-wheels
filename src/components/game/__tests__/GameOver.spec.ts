import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import GameOver from "../GameOver.vue";

import vuetify from "@/plugins/vuetify";

const pushedRoutes: string[] = [];
const mockRouter = {
  push: (route: string) => {
    pushedRoutes.push(route);
  },
};

function mountComponent() {
  return mount(GameOver as any, {
    global: {
      plugins: [vuetify],
      mocks: {
        $router: mockRouter,
      },
    },
  });
}

describe("GameOver", () => {
  it("renders full component", () => {
    const wrapper = mountComponent();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("clicking on Play Again redirects to the init view", async () => {
    const wrapper = mountComponent();

    await wrapper.find(".v-btn").trigger("click");

    expect(pushedRoutes).toContain("init");
  });
});
