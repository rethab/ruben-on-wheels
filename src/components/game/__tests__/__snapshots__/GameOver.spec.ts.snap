// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameOver > renders full component 1`] = `
"<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <div class=\\"v-progress-linear__indeterminate\\">
          <div class=\\"v-progress-linear__indeterminate long\\"></div>
          <div class=\\"v-progress-linear__indeterminate short\\"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-title\\">Game Over</div>
  <div class=\\"v-card-text\\">Unfortunately, nobody is going to visit Ruben this time ðŸ¥²</div>
  <div class=\\"v-card-actions\\">
    <div class=\\"flex-grow-1\\"></div><button type=\\"button\\" class=\\"v-btn v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-tonal\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><span class=\\"v-btn__prepend\\"><i class=\\"mdi-play mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span><span class=\\"v-btn__content\\" data-no-activator=\\"\\">Play Again</span>
      <!---->
      <!---->
    </button>
  </div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;
